vllm/distributed/parallel_state.py

```diff
@@ -433,20 +433,42 @@ def gather(self,
         if dim < 0:
             # Convert negative dim to positive.
             dim += input_.dim()
-        # Allocate output tensor.
-        if self.rank_in_group == dst:
-            gather_list = [torch.empty_like(input_) for _ in range(world_size)]
-        else:
-            gather_list = None
-        # Gather.
-        torch.distributed.gather(input_,
-                                 gather_list,
-                                 dst=self.ranks[dst],
-                                 group=self.device_group)
-        if self.rank_in_group == dst:
-            output_tensor = torch.cat(gather_list, dim=dim)
+
+        if current_platform.is_xpu():
+            input_size = input_.size()
```

Comment:
pleas add some comments why we use `all_gather_into_tensor` to implement gather.